library/azure_rm_devtestlabvirtualmachine.py

```diff
@@ -0,0 +1,533 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_devtestlabvirtualmachine
+version_added: "2.8"
+short_description: Manage Azure DevTest Lab Virtual Machine instance.
+description:
+    - Create, update and delete instance of Azure DevTest Lab Virtual Machine.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    lab_name:
+        description:
+            - The name of the lab.
+        required: True
+    name:
+        description:
+            - The name of the virtual machine.
+        required: True
+    notes:
+        description:
+            - The notes of the virtual machine.
+    os_type:
+        description:
+            - Base type of operating system.
+        choices:
+            - windows
+            - linux
+        default: linux
+    vm_size:
+        description:
+            - A valid Azure VM size value. For example, 'Standard_D4'. The list of choices varies depending on the
+              subscription and location. Check your subscription for available choices. Required when creating a VM.
+    user_name:
+        description:
+            - The user name of the virtual machine.
+    password:
+        description:
+            - The password of the virtual machine administrator.
+    ssh_key:
+        description:
+            - The SSH key of the virtual machine administrator.
+    lab_subnet:
+        description:
+            - An existing subnet within lab's virtual network
```

Comment:
i think you mentioned that it must be subnet in same lab, so name is enough. why you change it finally?